istioctl/cmd/kubeinject.go

```diff
@@ -354,20 +514,15 @@ kube-inject on deployments to get the most up-to-date changes.
 	injectCmd.PersistentFlags().StringVar(&valuesFile, "valuesFile", "",
 		"injection values configuration filename.")
 
-	injectCmd.PersistentFlags().BoolVar(&emitTemplate, "emitTemplate", false,
-		"Emit sidecar template based on parameterized flags")
-	_ = injectCmd.PersistentFlags().MarkHidden("emitTemplate")
-
 	injectCmd.PersistentFlags().StringVarP(&inFilename, "filename", "f",
 		"", "Input Kubernetes resource filename")
 	injectCmd.PersistentFlags().StringVarP(&outFilename, "output", "o",
 		"", "Modified output Kubernetes resource filename")
 
 	injectCmd.PersistentFlags().StringVar(&meshConfigMapName, "meshConfigMapName", defaultMeshConfigMapName,
 		fmt.Sprintf("ConfigMap name for Istio mesh configuration, key should be %q", configMapKey))
-	injectCmd.PersistentFlags().StringVar(&injectConfigMapName, "injectConfigMapName", defaultInjectConfigMapName,
-		fmt.Sprintf("ConfigMap name for Istio sidecar injection, key should be %q.", injectConfigMapKey))
-
```

Comment:
My comment was that you should mark `--injectConfigMapName` as hidden the way emitTemplate was, and/or add " [DEPRECATED]" to it's description.  We can't remove it until it has spent a version in the deprecated stage.

Reply:
fixed